ch3/docker-compose.yaml

```
@@ -0,0 +1,59 @@
+version: "3.7"
+services:
+  backend-store:
+    image: postgres:14.0
+    container_name: backend-store
+    environment:
+      POSTGRES_USER: myuser
+      POSTGRES_PASSWORD: mypassword
+      POSTGRES_DB: mydatabase
+      TZ: Asia/Seoul
+    ports:
+      - 5432:5432
+    healthcheck:
+      test: ["CMD", "pg_isready", "-q", "-U", "myuser", "-d", "mydatabase"]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+
+  artifact-store:
+    image: minio/minio
+    container_name: artifact-store
+    working_dir: /usr/app
+    ports:
+      - 9000:9000
+      - 9001:9001
+    environment:
+      MINIO_ROOT_USER: mystorage
+      MINIO_ROOT_PASSWORD: mystoragepw
+    command: server /usr/app --console-address :9001
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
+      interval: 30s
+      timeout: 20s
+      retries: 3
+
+  mlflow:
+    platform: linux/amd64
+    container_name: mlflow
+    build:
+      context: .
+      dockerfile: ./Dockerfile
+    ports:
+      - 5000:5000
+    environment:
+      AWS_ACCESS_KEY_ID: "mystorage"
+      AWS_SECRET_ACCESS_KEY: "mystoragepw"
+    command:
+      - /bin/sh
+      - -c
+      - |
+        mc config host add mlflowminio http://artifact-store:9000 mystorage mystoragepw &&
+        mc mb --ignore-existing mlflowminio/mlflow
+        mlflow server \
+        --backend-store-uri postgresql://myuser:mypassword@backend-store/mydatabase \
+        --default-artifact-root s3://mlflow/ \
+        --host 0.0.0.0
+    depends_on:
+      - backend-store
+      - artifact-store
```

```suggestion
    depends_on:
      backend-store:
        condition: service_healthy
      artifact-store:
        condition: service_healthy
```
reference: https://github.com/peter-evans/docker-compose-healthcheck#waiting-for-postgresql-to-be-healthy

---

반영 했습니다 :)